Library/Homebrew/software_spec.rb

```diff
@@ -53,7 +53,7 @@ def owner=(owner)
     @resource.owner = self
     resources.each_value do |r|
       r.owner     = self
-      r.version ||= version
+      r.version ||= version.dup
```

Comment:
I would suggest `version.head? ? HeadVersion.new("HEAD") : version`

As the commit of formula should have nothing to do with the version of resources.